data/event/event.go

```diff
@@ -19,42 +24,44 @@ const (
 //     "transmission": "manual",
 //   },
 // }
+
+// Events stores events
+type Events []*Event
+
+// Event defines the structure of an event
 type Event struct {
-	Summary  string `json:"summary"`
-	Category string `json:"category,omitempty"`
+	Timestamp int64  `json:"timestamp"`
+	Summary   string `json:"summary"`
+	Category  string `json:"category,omitempty"`
 	// Attributes are optional, they represent additional information that
 	// can be attached to an event.
 	Attributes map[string]interface{} `json:"attributes,omitempty"`
 }
 
 // New creates a new event.
-func New(summary, category string) *Event {
+func New(timestamp time.Time, summary, category string) (*Event, error) {
+	if len(summary) == 0 {
+		return nil, err.ParameterCannotBeEmpty("summary")
+	}
 	return &Event{
+		Timestamp:  timestamp.Unix(),
 		Summary:    summary,
 		Category:   category,
 		Attributes: make(map[string]interface{}),
-	}
+	}, nil
 }
 
 // NewNotification creates a new notification event.
-func NewNotification(summary string) *Event {
-	return New(summary, NotificationEventCategory)
-}
-
-// NewWithAttributes creates a new event with the given attributes
-func NewWithAttributes(summary, category string, attributes map[string]interface{}) *Event {
-	e := New(summary, category)
-	e.Attributes = attributes
-	return e
-}
-
-func (e *Event) setAttribute(key string, val interface{}) {
-	e.Attributes[key] = val
+func NewNotification(summary string) (*Event, error) {
+	return New(time.Now(), summary, NotificationEventCategory)
 }
 
-// AddCustomAttributes add customAttributes to the Event
-func AddCustomAttributes(e *Event, customAttributes []attribute.Attribute) {
-	for _, attr := range customAttributes {
-		e.setAttribute(attr.Key, attr.Value)
+// AddAttribute adds an attribute to the Event
```

Comment:
how can one add attributes/fields to an event out of the `attributes` set?

infra agent converts these into `"attr.*"` keys, which isn't probably what users want/expect

maybe we should remove this prefix thing done by the agent?

Reply:
I think we prefix them in the agent to avoid collision between collected data and custom attributes